### PR TITLE
faust 1 and 2 update, plus adapt affected pkgs

### PR DESCRIPTION
**Pull request by magnetophon**
###### Motivation for this change

Sorry for the many changes, but they are all kind of inter-related: faust got a new library organization, so all faust code needs to be updated.

I figured it made sense to also put my faust DSP in it's own sub-category, as requested  [here](https://github.com/NixOS/nixpkgs/pull/17215#issuecomment-235058927).

I also added the 2 new DSP's that originated that request in https://github.com/NixOS/nixpkgs/pull/17215.
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by sternenseemann** [comment]
You should probably update the commit messages to follow the [guidelines described in CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) and split the changes into two commits (faust1 and faust2).

**Review by sternenseemann — COMMENTED** [review]
This should also be rebased to be included in the other commit(s).

**Comment by magnetophon** [comment]
@sternenseemann Thanks. I will, once I make sure hydra passes. I've only buillt it on ``nixos-stable''.

**Comment by magnetophon** [comment]
@sternenseemann Are these commits OK?

**Comment by sternenseemann** [comment]
You could integrate the library updates into the commits, but it is fine like this as well I guess.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -13594,6 +13586,19 @@ in
 
   lyx = callPackage ../applications/misc/lyx { };
 
+  magnetophonDSP = {
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: now you do want to group them like this? 
https://github.com/NixOS/nixpkgs/pull/17215#issuecomment-235880114

**Comment by magnetophon** [comment]
@FRidh like this?

**Comment by sternenseemann** [comment]
@magnetophon he was referring to [your comment](https://github.com/NixOS/nixpkgs/pull/17215#issuecomment-235880114)

**Comment by magnetophon** [comment]
@sternenseemann sorry, I don't get it.

**Comment by FRidh** [comment]
@magnetophon in the earlier PR that I referred to you eventually choose not to group and include the items.

**Comment by magnetophon** [comment]
@FRidh So I should not group them?
Or not include them at all? I would like them to be in nixpkgs if that's OK.

In the other PR I backtracked because not all of them had a decent readme and such, but in the meantime I've come to rely on these pkgs, so I fixed the flaws and did another PR.

What do you suggest? And be explicit please :)

File: pkgs/applications/audio/magnetophonDSP/faustCompressors/default.nix [hunk 2]
```diff
@@ -1,22 +1,28 @@
-{ stdenv, fetchFromGitHub, faust2jaqt, faust2lv2gui }:
+{ stdenv, fetchFromGitHub, faust2jaqt, faust2lv2 }:
 stdenv.mkDerivation rec {
   name = "faustCompressors-${version}";
-  version = "0.1.1";
+  version = "v1.1.1";
```
Comment (FRidh) [inline pkgs/applications/audio/magnetophonDSP/faustCompressors/default.nix]: this v should not be here

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by FRidh** [comment]
@magnetophon it's fine to have them here, and to have them grouped like this. I was just surprised to see this grouping because of the previous PR.

**Comment by magnetophon** [comment]
@FRidh happy to hear that.
The v has moved.

**Comment by magnetophon** [comment]
@FRidh @sternenseemann 
travis hits the 1000 line limit, but seems fine otherwise.

**Comment by magnetophon** [comment]
@FRidh @sternenseemann 
I looked a bit closer, and the problem is that travis wants to build qt-4.8.7.
I don't see why though.

**Comment by FRidh** [comment]
@magnetophon you can ignore travis since its quite unreliable.
Could you rebase this PR. Everything else is ok.

**Comment by magnetophon** [comment]
@FRidh Done.

**Comment by magnetophon** [comment]
@FRidh  thanks.

**Pull request merged**